wasm.md

```diff
@@ -1531,6 +1496,238 @@ Start Function
          </moduleInst>
 ```
 
+Export
+------
+
+Exports make functions, tables, memories and globals available for importing into other modules.
+
+```k
+    syntax Defn       ::= ExportDefn
+    syntax ExportDefn ::= "(" "export" String "(" Externval ")" ")"
+ // ---------------------------------------------------------------
+    rule <k> ( export ENAME ( _:AllocatedKind TFIDX:TextFormatIdx ) ) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <exports> EXPORTS => EXPORTS [ ENAME <- TFIDX ] </exports>
+           ...
+         </moduleInst>
+```
+
+Exports can also be declared like regular functions, memories, etc., by giving an inline export declaration.
+In that case, it simply desugars to an export followed by the definition, after introducing a fresh identifier if no identifier is present.
+Note that it is possible to define multiple exports inline, i.e., export a single entity under many names.
+
+```k
+    syntax InlineExport  ::= "(" "export" String ")"
+ // ------------------------------------------------
+
+    syntax GlobalSpec ::= InlineExport GlobalSpec
```

Comment:
I am not a fan of using those `XXXspec` to integrate the normal form of definitions and the abbreviated form of definitions. It makes the coupling between definitions pretty heavy and makes the logic complicated. I prefer to have:
```
syntax InlineDecls ::= InlineExport InlineDecls
syntax GlobalDefn ::= "(" "global" OptionalId InlineExport InlineDecls GlobalSpec ")"
// ---
rule <k> ( global                  EXPO:InlineExport InlineDecls SPEC:GlobalSpec )
          => ( global #freshId(NEXTID) EXPO         InlineDecls      SPEC            )
          ...
         </k>
         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
```
But this is just my personal opinion. : )

Reply:
Now that you mention it, the exports aren't really part of the "specification" of globals/functions/tables/mems, so yeah, that's a bit weird.

What would `InlineDecls` be in this case? What makes it different? Would it be like what we had before, with a list type `InlineExports`, only that it's non-empty? So it could also be something like this:

```
    syntax InlineExport   ::= "(" "export" String ")"
    syntax InlineExports  ::= InlineExport | InlineExport InlineExports
 // -------------------------------------------------------------------

    syntax GlobalDefn ::= "(" "global" OptionalId InlineExports GlobalSpec ")"
 // ---------------------------------------------
 ...
```

The problem is that we get more rules: one to deal with the base case (singleton list), one to deal with the inductive case, and one to introduce identifiers. So in total, we get more rules: one for functions, one for tables, etc.

My thinking here was that using `syntax GlobalSpec ::= InlineExport GlobalSpec` we get kinda close to the spec, where they give these inline exports (see page 118 for example) as `‘(’ ‘func’ id? ‘(’ ‘export’ name ‘)’ ... ‘)’` and the explanation "The latter abbreviation can be applied repeatedly, with “. . .” containing another import or export."

Maybe there's a better naming than `*Spec`?

Reply:
Yes this will introduce more rules. Maybe using `GlobalSpec` as you did is fine. Better to get confirmation from @ehildenb